delegatedrouting/listener.go

```diff
@@ -486,6 +496,42 @@ func contextIDToStr(contextID []byte) string {
 	return base64.StdEncoding.EncodeToString(contextID)
 }
 
+func (listener *Listener) flushWorker(ctx context.Context) {
+	var t *time.Timer
+
+	flushFunc := func() {
+		// we don't want flush to happen while re-provide is running
+		listener.lock.Lock()
+		defer listener.lock.Unlock()
+		// flush only if the current chunk has some cids in it and the time since the current chunk has been created is greater than the flush frequency
+		if len(listener.chunker.currentChunk.Cids) > 0 &&
+			time.Since(listener.chunker.currentChunkTime) > listener.adFlushFrequency {
+			err := listener.chunker.flushCurrentChunk(ctx, func(cc *cidsChunk) error {
+				return listener.notifyPutAndPersist(ctx, cc)
+			})
+			if err != nil {
+				log.Warnw("Error flushing current chunk", "err", err)
+			}
+		}
+	}
+
+	for {
+		select {
+		case <-ctx.Done():
+			return
+		default:
+			flushFunc()
+		}
+
+		t = time.NewTimer(listener.adFlushFrequency)
```

Comment:
Create `time.Ticker` once and reuse. This creates a ticker on every iteration.

Reply:
Good point - fixing